### PR TITLE
[inductor][scheduler] Use set for origin

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119861

xref - https://github.com/pytorch/pytorch/issues/119440

This avoids node > node comparison if the origin order is same in the origins tuple. However, I am unable to come up with a test case where this could happen.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -2248,9 +2248,9 @@ def get_order(n):
                 self.origin_to_index.update({n: i for i, n in enumerate(n.graph.nodes)})
             return self.origin_to_index[n]
 
-        origins = [(get_order(e), e) for n in node.get_nodes() for e in n.node.origins]
+        origins = {(get_order(e), e) for n in node.get_nodes() for e in n.node.origins}
         if origins:
-            _, last = max(origins)
+            _, last = max(origins, key=lambda x: x[0])
```
Comment (Skylion007) [inline torch/_inductor/scheduler.py:2253]:
```suggestion
            _, last = max(origins, key=operator.itemgetter(0))
```

**Review by Skylion007 — APPROVED** [review]

**Review by eellison — APPROVED** [review]
Nit: use ordered set https://github.com/pytorch/pytorch/blob/ee3a7bdc2d0966ab788e48e640879fcf4cd1c69e/torch/_inductor/fx_passes/group_batch_fusion.py#L755

File: torch/_inductor/scheduler.py [hunk 2]
```diff
@@ -2244,9 +2245,13 @@ def get_order(n):
                 self.origin_to_index.update({n: i for i, n in enumerate(n.graph.nodes)})
             return self.origin_to_index[n]
 
-        origins = [(get_order(e), e) for n in node.get_nodes() for e in n.node.origins]
+        # Use a dict to have ordering
+        origins = {
```
Comment (Skylion007) [inline torch/_inductor/scheduler.py:2249]: Use `dict.fromkeys()` instead here.

**Pull request closed without merging**